astroid/brain/brain_pathlib.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2022 Deepyaman Datta <deepyaman.datta@utexas.edu>
+
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
+# For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
+
+from typing import Iterator, Optional
+
+from astroid import bases, context, inference_tip
+from astroid.const import PY310_PLUS
+from astroid.exceptions import UseInferenceDefault
+from astroid.manager import AstroidManager
+from astroid.nodes.node_classes import Attribute, Slice, Subscript
+from astroid.nodes.node_ng import NodeNG
+
+
+def _looks_like_parents_subscript(node: NodeNG) -> bool:
+    return (
+        isinstance(node, Subscript)
+        and isinstance(node.value, Attribute)
+        and node.value.attrname == "parents"
+    )
+
+
+def infer_parents_subscript(
+    subscript_node: Subscript, ctx: Optional[context.InferenceContext] = None
+) -> Iterator[bases.Instance]:
+    if isinstance(subscript_node.slice, Slice):
+        raise UseInferenceDefault
+
+    yield from subscript_node.value.expr.infer(context=ctx)
```

Comment:
As mentioned earlier, it's not quite enough to check that `attrname == "parents"`.
A rough idea what might work instead
```py
    inferred = next(subscript_node.value.expr.infer(context=ctx))
    if not (
        isinstance(inferred, bases.Instance)
        and isinstance(inferred._proxied, nodes.ClassDef)
        and inferred.qname().startswith("pathlib.")
        and inferred.qname().endswith("Path")
    ):
        raise UseInferenceDefault
    if isinstance(subscript_node.slice, Slice):
        raise UseInferenceDefault

    return iter([inferred])
```

Reply:
I think the check is somewhat unnecessary because `subscript_node.value` is `nodes.Attribute` due to `_looks_like_parents_subscript`.
However, we don't really "know" this when we are in this function and therefore once we turn on `mypy` it will (rightly) complain that `qname` is not defined on `value`. I think it makes sense to add these `isinstance` checks as well.

Reply:
@cdce8p @DanielNoord I have updated this to add the `isinstance` checks to my implementation.